lib/ui/compositing.dart

```diff
@@ -398,6 +407,27 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   }
   EngineLayer _pushColorFilter(_ColorFilter filter) native 'SceneBuilder_pushColorFilter';
 
+  /// Pushes an image filter operation onto the operation stack.
+  ///
+  /// The given filter is applied to the children's rasterization before compositing them into
+  /// the scene.
+  ///
+  /// {@macro dart.ui.sceneBuilder.oldLayer}
+  ///
+  /// {@macro dart.ui.sceneBuilder.oldLayerVsRetained}
+  ///
+  /// See [pop] for details about the operation stack.
+  ImageFilterEngineLayer pushImageFilter(ImageFilter filter, { ImageFilterEngineLayer oldLayer }) {
+    assert(filter != null);
+    assert(_debugCheckCanBeUsedAsOldLayer(oldLayer, 'pushImageFilter'));
+    final _ImageFilter nativeFilter = filter._toNativeImageFilter();
+    assert(nativeFilter != null);
+    final ImageFilterEngineLayer layer = ImageFilterEngineLayer._(_pushImageFilter(nativeFilter));
+    assert(_debugPushLayer(layer));
+    return layer;
+  }
+  EngineLayer _pushImageFilter(_ImageFilter filter) native 'SceneBuilder_pushImageFilter';
+
   /// Pushes a backdrop filter operation onto the operation stack.
   ///
   /// The given filter is applied to the current contents of the scene prior to
```

Comment:
please also update the docs in other places, e.g. ImageFilter currently references pushBackdropFilter but should also mention pushImageFilter

Reply:
Good point.

Reply:
There is no filter for this yet, though. Do I pre-announce the ImageFiltered that isn't in the framework yet?

Reply:
@Hixie how do we handle the chicken/egg issue of mentioning forthcoming widgets in the dart:ui stuff in the engine repo?  Do I "push impl, roll to framework, push widget, then go back and update the ImageFilter docs"? Or is it OK to mention it as if it exists even though that won't happen until we get everything set for it and go through the PR reviews?

Reply:
Done. The reference won't make sense until I push the framework part of the fix, but doing it now avoids an extra PR.